src/Capgemini.PowerApps.PackageDeployerTemplate/Services/MailboxDeploymentService.cs

```diff
@@ -0,0 +1,162 @@
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Update queue's email address.
+        /// </summary>
+        /// <param name="mailboxConfig">The mailbox config to use.</param>
+        /// <returns>Returns Queue ID.</returns>
+        private Guid UpdateEmailAddress(MailboxConfig mailboxConfig)
+        {
+            var retrieveMultipleResponse = this.crmSvc.RetrieveMultipleByAttribute(Constants.Mailbox.Queue.LogicalName, Constants.Mailbox.Queue.Fields.EmailAddress, new object[] { mailboxConfig.SourceEmailaddress });
+            var entity = retrieveMultipleResponse != null ? retrieveMultipleResponse.Entities.FirstOrDefault() : null;
+            if (entity == null)
+            {
+                this.logger.LogInformation($"No queue exist with emailaddress:{mailboxConfig.SourceEmailaddress}.");
+                return Guid.Empty;
+            }
+
+            entity[Constants.Mailbox.Queue.Fields.EmailAddress] = mailboxConfig.TargetEmailaddress;
+            this.crmSvc.Update(entity);
+            return entity.Id;
+        }
+
+        /// <summary>
+        /// Approve the Email.
+        /// </summary>
+        /// <param name="entityid">Queue ID to use.</param>
+        /// <returns>Returns approval status(true/false).</returns>
+        private bool ApproveEmail(Guid entityid)
+        {
+            bool isApproved = true;
+            Entity updateEntity = new Entity(Constants.Mailbox.Queue.LogicalName, entityid);
+            updateEntity.Attributes.Add(new KeyValuePair<string, object>(Constants.Mailbox.Queue.Fields.EmailrouterAccessApproval, new OptionSetValue((int)EmailRouterAccessApproval.Approved)));
+            this.crmSvc.Update(updateEntity);
+
+            var entity = this.crmSvc.Retrieve(updateEntity.LogicalName, entityid, new ColumnSet(Constants.Mailbox.Queue.Fields.EmailrouterAccessApproval));
+
+            int i = 0;
+            int count = 1;
+            int delay = 30;
+            while (entity.GetAttributeValue<OptionSetValue>(Constants.Mailbox.Queue.Fields.EmailrouterAccessApproval).Value != (int)EmailRouterAccessApproval.Approved)
+            {
+                i++;
+                if (i > count)
+                {
+                    this.logger.LogInformation("[Failure] Email Address not Approved within allotted timeframe.");
+                    break;
+                }
+
+                Thread.Sleep(1000 * delay);
+                entity = this.crmSvc.Retrieve(updateEntity.LogicalName, entityid, new ColumnSet(Constants.Mailbox.Queue.Fields.EmailrouterAccessApproval));
+            }
+
+            if (entity.GetAttributeValue<OptionSetValue>(Constants.Mailbox.Queue.Fields.EmailrouterAccessApproval).Value != (int)EmailRouterAccessApproval.Approved)
+            {
+                this.logger.LogInformation("[Failure] Email Address not Approved.");
```

Comment:
Should be using `LogWarning` or `LogError` here rather than `LogInformation` with `[Failure]` in the message.